models/workload.go

```diff
@@ -172,6 +172,13 @@ type WorkloadListItem struct {
 
 	// Names of the waypoint proxy workloads, if any
 	WaypointWorkloads []string `json:"waypointWorkloads"`
+
+	// ValidationKey is a pre-calculated key string: "cluster:namespace:name"
+	ValidationKey string
```

Comment:
Can this be an object similar to `IstioValidationKey` to reduce the risk of getting the formatting right every time with the string? It looks like the comment is already out of date since the key contains `Kind` as well. I don't think any of the fields (cluster,namespace,name,kind) allow `:` in the values but if they did that would also cause issues.

Reply:
I'm not sure I want to make this an Object, the idea is to pre-calculate an identifying string one time when the workload is pulled in.  It will not be parsed, it's only used as the key in the changeMap to identify this workload in the future.  I'm fairly sure there is only one workload with the cluster:namespace:name combo but if you think it needs to be more complex I guess we could add to it, or perform a hash of various info.

`Kind` is not currently used in this identifier, it's set as described in `WorkloadsService.fetchWorkloadsFromCluster`.  Maybe you saw the keys being generated for the Istio config?  Those are not current pre-calculated, and they do have a hacky "Kind" that I made up.

Reply:
Then having a some kind of method to construct the key would be nice. Something like `validations.ValidationWorkloadKey(cluster, ...)`. Ideally workload parsing would know nothing about the internals of validations since validations are independent of parsing a workload but it's currently a mix. We can keep it as is for the time being.

Reply:
Not unlike my above comment, I like the cleanliness and separation of concern.  But to some degree that ship sailed.  We have a bunch of calculated information hanging off of the Workloads.  I mean, WorkloadListItem is horrible, it's actually a server-side construct designed to support a UI list page.  Still, in another PR maybe we can improve this validation key stuff, and not keep piling on.  Like you say, we could at least maybe extract the key generation logic, if not the storage location.